ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/HardFork/Combinator/Ledger.hs

```diff
@@ -1110,6 +1112,44 @@ class ( Show (HardForkTxOut xs)
   default txOutEjections :: CanHardFork xs => NP (K (NS WrapTxOut xs) -.-> WrapTxOut) xs
   txOutEjections = composeTxOutTranslations $ ipTranslateTxOut hardForkEraTranslation
 
+instance (CanHardFork xs, HasHardForkTxOut xs) => CanUpgradeLedgerTables (LedgerState (HardForkBlock xs)) where
+  upgradeTables
+    (HardForkLedgerState (HardForkState hs0))
+    (HardForkLedgerState (HardForkState hs1))
+    (LedgerTables (ValuesMK vs)) =
+        LedgerTables
+      $ ValuesMK
+      $ (
+    if (nsToIndex $ Telescope.tip hs0) /= (nsToIndex $ Telescope.tip hs1)
```

Comment:
I think we can write something more direct by recursing on both telescopes. Or maybe something like this already exists, like `matchTelescope`.

It's maybe not a super important optimisation to apply, what do you think? We'll be calling this function every time we push diffs

Reply:
Done with `telescopesMismatch`